### ISSUE TITLE
Human-friendly ghc option names

### ISSUE THREAD

**Issue by mitchellwrosen** [haskell/cabal#6071]
It would be nice to be able to write something like

```
ghc-options:
  threaded-runtime: true
  warnings: all
  runtime-options:
    hec: N
    parallel-gc-hecs: 2
    nursery-size: 64m
```

instead of the usual

```
ghc-options:
  -threaded -Wall -with-rtsopts=-N -A64m -qn2
```

**Comment by 23Skidoo** [comment]
I'm -1 on this, it introduces an additional Cabal-specific vocabulary for the concepts users already know by different names and will have to relearn, plus extra implementation complexity that Cabal maintainers will have to deal with. Plus it becomes more wordy, and I can 100% guarantee that a lot of people will not like it.

So basically a lot of cons and few (if any) pros. The feature is purely cosmetic and doesn't IMO pay for itself.

**Comment by mitchellwrosen** [comment]
Human-readable names for GHC options would simply make the overall experience of using this tool nicer for beginners. 

Of course some people would prefer to write `-N2` rather than `num-capabilities: 2`, and giving a longer, better name to _every_ GHC option would be a massive effort - so I was not trying to suggest that cabal would no longer allow you to pass literal options to GHC at all (though that was not clear from the original post).

Some options, especially those that tweak the parallel garbage collector, have names that differ only by a single letter (`-qg`, `-qb`, `-qn`). Since cabal takes a high-level definition of a package and invokes GHC many times all of the appropriate low-level options, it makes some sense (to me) to also take on the responsibility of compiling highish-level GHC concepts (like the size of the nursery) into low-level flags.

So I'm personally still +1 on this issue, despite your very valid points about maintenance burden and having to come up with custom, Cabal-specific names.

**Comment by 23Skidoo** [comment]
Sorry, I'm still unconvinced. "Nicer for beginners" is an especially weak argument: having two names for everything is confusing (you need to consult a GHC-to-Cabal dictionary every time you want to change something) and "nicer" names like `hec: N` won't really tell beginners anything, they will still have to go to GHC documentation to learn these concepts. 

Second, you will also need to maintain a second version of documentation for stuff that is already documented in the GHC manual. It will have to be updated with each GHC release and will inevitably bitrot. 

Third, for RTS options like `-qb` that you mentioned, people usually tweak them in the production binary to find optimal values (so they use official names), and if they later decide to bake them in, they just copy-paste the command line to `ghc-options`. So all the toil you poured into implementing this will be in vain.

So basically, this feature is not user-friendly (people will need to learn new names for things they're already familiar with), a lot of work to implement and maintain, and will most likely be only used by a small minority.

I suggest adding alternative aliases for `-qb` and other cryptically-named options to GHC itself instead.

**Comment by tom-audm** [comment]
> I suggest adding alternative aliases for -qb and other cryptically-named options to GHC itself instead.

I'd also support this!

**Comment by mitchellwrosen** [comment]
@23Skidoo No worries, good points! Hope I did not come off as demanding. I will close this issue now.
